src/components/marquee-wizard/marquee-wizard.component.spec.ts

```diff
@@ -273,3 +281,108 @@ describe('Marquee wizard with delayed step creation', () => {
         await fixture.whenStable();
     }
 });
+
+
+interface WizardStep {
+    header: string;
+    content: string;
+}
+
+@Component({
+    selector: 'wizard-visited-change-test-app',
+    template: `
+        <ux-marquee-wizard
+            (stepChanging)="stepChanging($event)"
+            (stepChange)="stepChange($event)"
+            (onNext)="onNext($event)"
+        >
+            <ux-marquee-wizard-step *ngFor="let step of steps"
+                [header]="step.header" [valid]="step.valid"
+                (visitedChange)="visitedChanged()"
+            >
+                <p>{{ step.content }}</p>
+                <button class="toggle-validity-button" (click)="step.valid = !step.valid">Toggle validity</button>
+            </ux-marquee-wizard-step>
+        </ux-marquee-wizard>
+    `
+})
+class MarqueeWizardVisitedChangeTestComponent {
+    steps: WizardStep[] = [
+        {
+            header: '1. First Step',
+            content: 'Content of step 1.',
+        },
+        {
+            header: '2. Second Step',
+            content: 'Content of step 2.',
+        },
+        {
+            header: '3. Third Step',
+            content: 'Content of step 3.',
+        },
+        {
+            header: '4. Fourth Step',
+            content: 'Content of step 4.',
+        }
+    ];
+    stepChanging(_: StepChangingEvent) {}
+    stepChange(_: number) {}
+    onNext(_: number) {}
+    visitedChanged(_: boolean) { }
+}
+
+describe('Marquee wizard with visitedChange event', () => {
+    let component: MarqueeWizardVisitedChangeTestComponent;
+    let fixture: ComponentFixture<MarqueeWizardVisitedChangeTestComponent>;
+    let nativeElement: HTMLElement;
+    let visitedChanged: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [MarqueeWizardModule, CommonModule],
+            declarations: [MarqueeWizardVisitedChangeTestComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(async () => {
+        fixture = TestBed.createComponent(MarqueeWizardVisitedChangeTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        visitedChanged = spyOn(component, 'visitedChanged');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+        await fixture.whenStable();
+    });
+
+    it('should trigger a visitedChange event when valid modified on the current step, when other steps ahead are visited', async () => {
```

Comment:
https://plnkr.co/edit/dnuBVz3Z9BIyfey9

When I try this here I don't see visitedChange getting emitted. So for example, I open the marquee wizard, hit next until I get to a step near the end. Now I navigate back to the second step, mark it as invalid. The other steps remain visited and don't seem to get reset?

Am I misunderstanding this? Or doing something wrong?